src/SpreadingDye.jl

```diff
@@ -33,74 +33,103 @@ function random_point_on_domain(m::AbstractMatrix{Bool})
     x
 end
 
-# grow the range one at a time
-function grow!(georange::AbstractMatrix{Bool}, edges::Set, dom::AbstractMatrix{Bool}, algo::Symbol; ignore_domain = false)
-    newcell = isempty(edges) ? random_point_on_domain(georange) : first(rand(edges))
-    while georange[newcell...]
-        isempty(edges) && push!(edges, jump(georange, dom, algo)) # allows for patchy ranges
-        edge, newcell = rand(edges)
-        pop!(edges, (edge, newcell))
+# choose a random new cell
+function _random_edge_or_jump(georange, edges, dom, algo)
+    while !isempty(edges)
+        newidx = rand(eachindex(edges))
+        newcell = popat!(edges, newidx)
+        !georange[newcell...] && return newcell
     end
-    georange[newcell...] = true
-    for nb in algos[algo]
+    return jump(georange, dom, algo)
+end
+
+# grow the range one at a time
+grow!(georange, edges, dom, algo::Symbol; ignore_domain = false) =
```

Comment:
I don't really understand the idea of this synonymization with `_grow`? Is this just another stylistic change?